execution-engine/src/wasi/common.rs

```diff
@@ -371,6 +371,21 @@ pub struct WasiWrapper {
     exit_code: Option<u32>,
     /// Whether clock functions (`clock_getres()`, `clock_gettime()`) should be enabled.
     pub enable_clock: bool,
+    /// Whether strace is enabled.
+    pub enable_strace: bool,
+}
+
+macro_rules! strace {
+    ($self: expr, $name: expr, $block: block) => {{
+        let result = (|| $block)();
+        if $self.enable_strace {
+            match result {
+                Ok(()) => eprintln!("{}(...) = Success", $name),
```

Comment:
How about calling `println!` here, or using the `log` crate? Should we consider system call errors as Veracruz errors?

Reply:
System call errors are fairly unusual but they are expected in some circumstances. I'm using `eprint` because Linux's `strace` writes to `stderr`: not an amazingly good argument, but I have to decide somehow.